packages/core/parcel/src/cli.js

```diff
@@ -464,6 +465,13 @@ async function normalizeOptions(
     })),
   ];
 
+  if (command.profileApplication) {
+    additionalReporters.unshift({
+      packageName: '@parcel/reporter-application-profiler',
+      resolveFrom: __filename,
```

Comment:
This means that `@parcel/reporter-application-profiler` needs to be a dependency of the `parcel` CLI.